test/saithrift/tests/sail3.py

```diff
@@ -1324,4 +1325,158 @@ def runTest(self):
+    v6_enabled = 1
+    ip_mask = '255.255.255.0'
+    addr_family = SAI_IP_ADDR_FAMILY_IPV4
+    lag_members = []
+    lags = []
+    lags_rifs = []
+    neighbors = []
+    routes = []
+    vr_id = 0
+    mac_action = SAI_PACKET_ACTION_FORWARD
+    src_port = 0
+    mac_pool = ['00:11:22:33:44:50',
+               '00:11:23:33:44:51',
+               '00:11:24:33:44:52',
+               '00:11:25:33:44:53',
+               '00:11:26:33:44:54',
+               '00:11:27:33:44:55',
+               '00:11:28:33:44:56',
+               '00:11:29:33:44:57',
+               '00:11:30:33:44:58',
+               '00:11:31:33:44:59',
+               '00:11:32:33:44:60',
+               '00:11:33:33:44:61',
+               '00:11:34:33:44:62',
+               '00:11:35:33:44:63',
+               '00:11:36:33:44:64',
+               '00:11:37:33:44:65',
+               '00:11:38:33:44:66']
+    
+    
+    def setup_lags(self, num_of_lags, port_list):
+        for i in xrange(num_of_lags):
+            self.lags.append(self.client.sai_thrift_create_lag([]))
+        for i in xrange(self.total_lag_port):
+            self.lag_members.append(sai_thrift_create_lag_member(self.client, self.lags[i % num_of_lags], port_list[i]))
+        for i in xrange(num_of_lags):
+            self.lags_rifs.append(sai_thrift_create_router_interface(self.client, self.vr_id, 1, self.lags[i], 0, self.v4_enabled, self.v6_enabled, ''))
+        for i in xrange(self.total_lag_port):
+            sai_thrift_create_neighbor(self.client, self.addr_family, self.lags_rifs[i%len(self.lags_rifs)], "10.10.%s.1" % str(i+1), self.mac_pool[i])
+            sai_thrift_create_route(self.client, self.vr_id, self.addr_family, "10.10.%s.0" % str(i+1), self.ip_mask, self.lags_rifs[i% len(self.lags_rifs)])
+
+    def teardown_lags(self, num_of_lags, port_list):
+        if (num_of_lags == 0 ): return
+        for i in xrange(self.total_lag_port):
+            sai_thrift_remove_neighbor(self.client, self.addr_family, self.lags_rifs[i%num_of_lags], "10.10.%s.1" % str(i+1), self.mac_pool[i])
+            sai_thrift_remove_route(self.client, self.vr_id, self.addr_family, "10.10.%s.0" % str(i+1), self.ip_mask, self.lags_rifs[i%num_of_lags])
+        for rif in self.lags_rifs:
+            self.client.sai_thrift_remove_router_interface(rif)
+        del self.lags_rifs[:]
+        for lag_member in self.lag_members:
+            self.client.sai_thrift_remove_lag_member(lag_member)
+        del self.lag_members[:]
+        for lag in self.lags:
+            self.client.sai_thrift_remove_lag(lag)
+        del self.lags[:]
+        
+    def send_and_verify_packets(self, num_of_lags, port_list):
+        exp_pkts = [0]*self.total_lag_port
+        pkt_counter = [0] * self.total_lag_port
+        destanation_ports = range(self.total_lag_port)
+        sport = 0x1234
+        dport = 0x50
+        src_mac = self.dataplane.get_mac(0, 16)
+        IP_LAST_WORD_RANGE = 254
+        IP_2ND_LAST_WORD_RANGE = 16
```

Comment:
In the worst case, when you have 1 lag with 16 members, each port only get 254/16 packets, then you need to vary the source IP in a larger range. for example i = 65536. and src ip = 10.0.i/255.i%255